src/Http/Middleware/PortalMiddleware.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace FluxErp\Http\Middleware;
+
+use Closure;
+use FluxErp\Models\Address;
+use FluxErp\Models\Order;
+use FluxErp\Models\OrderPosition;
+use FluxErp\Models\SerialNumber;
+use FluxErp\Models\Ticket;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Relations\Relation;
+use Illuminate\Http\Request;
+
+class PortalMiddleware
+{
+    public function handle(Request $request, Closure $next): mixed
+    {
+        if (request()->isPortal()) {
+            resolve_static(SerialNumber::class, 'addGlobalScope', [
+                'scope' => 'portal',
+                'implementation' => function (Builder $query) {
+                    $query->whereHas(
```

Comment:
group whereHas() and orWhereHas() together inside one where().
Otherwise there is a singular orWhereHas() in the query which might ruin other constraints.